targets/TARGET_ARM_FM/TARGET_FVP_MPS2/us_ticker.c

```diff
@@ -61,12 +87,10 @@ void us_ticker_set_interrupt(timestamp_t timestamp)
         us_ticker_init();
     }
 
-    uint32_t delta = timestamp - us_ticker_read();
-    // enable interrupt
-    US_TICKER_TIMER1->TimerControl = 0x0; // disable timer
-    US_TICKER_TIMER1->TimerControl = 0x62; // enable interrupt and set to 32 bit counter and set to periodic mode
-    US_TICKER_TIMER1->TimerLoad = (delta) * 25; //initialise the timer value
-    US_TICKER_TIMER1->TimerControl |= 0x80; //enable timer
+    uint32_t delta =  timestamp - us_ticker_read();
+    US_TICKER_TIMER2->TimerLoad = delta;                               // Set TIMER2 load value
```

Comment:
Can we safety change timer value while timer is running?

Reply:
I think we should also consider case when `delta` is equal to 0. Will interrupt be fired?

Reply:
In the current implementation, if 0 is set, interrupt will be fired.
I didn't find clear definition HAL API what happen if current_time is set to interrupt. but seems [This line](https://github.com/ARMmbed/mbed-os/blob/master/hal/us_ticker_api.h#L69) indicates a interrupt will need to be fired